src/main/java/com/example/demo/infrastructure/spring/data/SpringUserService.java

```diff
@@ -7,17 +7,18 @@
 import org.slf4j.LoggerFactory;
 
 import com.example.demo.application.exception.NotFoundException;
-import com.example.demo.infrastructure.repository.UserRepository;
+import com.example.demo.domain.User;
+import com.example.demo.domain.UserService;
 
 
 // TODO: Vote in which layer we do the mapping
-public class UserService {
+public class SpringUserService implements UserService {
```

Comment:
```suggestion
@Service
public class SpringUserService implements UserService {
```
Jag blir lite fundersam på namngivningsstrategin här – är tanken att vi ska använda prefixet 'Spring' på alla våra services? Jag har inte stött på det tidigare, så det vore intressant att höra hur du har tänkt här!

Reply:
Som jag varit inne på i tidigare kommentarer så döljer interfacet `UserService` implementationsdetaljer som till exempel vilken teknologi vi använder för att lagra information persistent. Klassen `SpringUserService` är en implementation som använder Spring Data för att lagra och hämta användare från en databas. Därför anser jag att `SpringUserService` är ett lämpligt namn. Jag tycker inte att vi ska ha interface med prefixet `I`, till exempel `IUserService`. Lika lite ska vi ha implementationsklasser med suffixet `Impl`, till exempel `UserServiceImpl`. Jag anser att ett bra interface höjer abstraktionsnivån och har alltid fler än en möjliga implementationer. Namnet på en implementationsklass kan (och bör?) antyda *hur* interfacet implementeras.